gui/src/3D/controls/slider3D.ts

```diff
@@ -0,0 +1,193 @@
+import { StandardMaterial } from "babylonjs/Materials/standardMaterial";
+import { Color3 } from "babylonjs/Maths/math.color";
+import { AbstractMesh } from "babylonjs/Meshes/index";
+
+export interface ISlider3DOptions {
+    minimum?: number,
+    maximum?: number,
+    value?: number
+};
+
+/**
+ * Class used to create a slider in 3D
+ */
+ export class Slider3D extends Control3D {
+    private _sliderBarMaterial: Material;
+    private _sliderThumbMaterial: Material;
+    private _sliderThumb: Mesh;
+    private _sliderBar: Mesh;
+
+    private _minimum: number;
+    private _maximum: number;
+    private _value: number;
+
+    /** Observable raised when the sldier value changes */
+    public onValueChangedObservable = new Observable<number>();
+
+    /**
+     * Creates a new slider
+     * @param name defines the control name
+     */
+     constructor(name?: string, options?: ISlider3DOptions) {
+        super(name);
+
+        this._minimum =  options?.minimum ? options.minimum : 0;
+        this._maximum = options?.maximum ? options.maximum : 100;
+        this.value = options?.value ? options.value : 50;       
+    }    
+
+    /** Gets or sets minimum value */
+    public get minimum(): number {
+        return this._minimum;
+    }
+
+    public set minimum(value: number) {
+        if (this._minimum === value) {
+            return;
+        }
+
+        this._minimum = Math.min(value, this._maximum);
+        this.value = this._value;
+    }
+
+     /** Gets or sets maximum value */
+     public get maximum(): number {
+         return this._maximum;
+     }
+ 
+     public set maximum(value: number) {
+         if (this._maximum === value) {
+             return;
+         }
+ 
+         this._maximum = Math.max(value, this._minimum);
+         this.value = this._value;
+     }
+
+    /** Gets or sets current value */
+    public get value(): number {
+        return this._value;
+    }
+
+    public set value(value: number) {
+        value = Math.max(Math.min(value, this._maximum), this._minimum);
+
+        if (this._value === value) {
+            return;
+        }
+
+        this._value = value;
+        this.onValueChangedObservable.notifyObservers(value);
+    }
+
+    protected get start(): number{
+        if(!this.node){
+            return -1;
+        }
+
+        return this._sliderBar.position.x - this._sliderBar.scaling.y / 2;
+    }
+
+    protected get end(): number{
+        if(!this.node){
+            return 1;
+        }
+
+        return this._sliderBar.position.x + this._sliderBar.scaling.y / 2;
+    }
+
+    // Mesh association
+    protected _createNode(scene: Scene): TransformNode {
+        const anchor = new AbstractMesh(`${this.name}_slider`, scene);    
+        
+        this._sliderBar = MeshBuilder.CreateCylinder(`${this.name}_sliderbar`, { diameter: 0.03, height: 1.0 }, scene);
+        this._sliderBar.rotation.z = -Math.PI / 2;
+        this._sliderBar.scaling.y = 2;

```

Comment:
>2 [](start = 36, length = 1)

should these arbitrary scale values be set through the options provided in the constructor? regardless, can we get any fallback values declared as nonexported constants at the top of the file/class?